test/integration/master/kube_apiserver_test.go

```diff
@@ -253,3 +296,27 @@ func TestReconcilerMasterLeaseMultiMoreMasters(t *testing.T) {
 func TestReconcilerMasterLeaseMultiCombined(t *testing.T) {
 	testReconcilersMasterLease(t, 3, 3)
 }
+
+type delayedCheck struct {
+	healthLock sync.Mutex
+	isHealthy  bool
+}
+
+func (h *delayedCheck) Name() string {
+	return "delayed-check"
+}
+
+func (h *delayedCheck) Check(req *http.Request) error {
+	h.healthLock.Lock()
+	defer h.healthLock.Unlock()
```

Comment:
If healthLock is RWMutex, we can take RLock in this func, making the check more efficient.